hoomd/hpmc/UpdaterBoxMC.cc

```diff
@@ -412,45 +415,47 @@ void UpdaterBoxMC::update(unsigned int timestep)
     hoomd::RandomGenerator rng(hoomd::RNGIdentifier::UpdaterBoxMC, m_seed, timestep);
 
     // Choose a move type
-    // This seems messy and can hopefully be simplified and generalized.
-    // This line will need to be rewritten or updated when move types are added to the updater.
-    Scalar range = m_volume_weight + m_ln_volume_weight + m_length_weight + m_shear_weight + m_aspect_weight;
-    if (range == 0.0)
+    auto const weight_total = m_weight_partial_sums.back();
+    if (weight_total == 0.0)
         {
-        // Attempt to execute with no move types set.
+        // Attempt to execute with all move weights equal to zero.
         m_exec_conf->msg->warning() << "No move types with non-zero weight. UpdaterBoxMC has nothing to do." << std::endl;
         if (m_prof) m_prof->pop();
         return;
         }
-    Scalar move_type_select = hoomd::detail::generate_canonical<Scalar>(rng) * range; // generate a number on (0, range]
+    auto const selected = hoomd::detail::generate_canonical<Scalar>(rng) * weight_total; // generate a number on (0, weight_total];
+    auto const move_type_select = std::distance(
```

Comment:
While I know what this is doing, I would consider making a small comment 1-2 lines briefly describing the selection process.

Reply:
I expanded the comment.